fvm/evm/types/balance.go

```diff
@@ -18,50 +26,82 @@ var (
 // But on the FLOW Vaults, we use Cadence.UFix64 to store values in Flow.
 // this could result in accidental conversion mistakes, the balance object here would
 // do the conversions and does appropriate checks.
+type Balance struct {
+	Value *big.Int
+}
+
+// NewBalance constructs a new balance from flow value (how its stored in Cadence Flow)
+func NewBalance(inp cadence.UFix64) *Balance {
```

Comment:
Is there any need to make `Balance` a pointer type? Maybe just keep it a value type?

Reply:
It doesn't have to be a pointer but I was not sure returning a value instead of a pointer might be a good idea given the underlying value uses pointers under the hood and would probably escape to heap anyway. 
But I'm not sure what is the best to use? what is your suggestion?

Reply:
@turbolent refactored to not use pointers and simplified the logic so it's less confusing.